bindings/nodejs/lib/client/client.ts

```diff
@@ -378,42 +402,60 @@ export class Client {
     }
 
     /**
-     * Fetch OutputResponse from given output IDs. Requests are sent in parallel.
+     * Finds an output by its ID and returns it as raw bytes.
+     * GET /api/core/v3/outputs/{outputId}
      */
-    async getOutputs(outputIds: OutputId[]): Promise<OutputResponse[]> {
+    async getOutputRaw(outputId: OutputId): Promise<Uint8Array> {
         const response = await this.methodHandler.callMethod({
-            name: 'getOutputs',
+            name: 'getOutputRaw',
             data: {
-                outputIds,
+                outputId,
             },
         });
 
-        const parsed = JSON.parse(response) as Response<OutputResponse[]>;
-        return plainToInstance(OutputResponse, parsed.payload);
+        return JSON.parse(response).payload;
     }
 
     /**
-     * Get outputs from provided output IDs (requests are sent
-     * in parallel and errors are ignored, can be useful for spent outputs)
-     *
-     * @param outputIds An array of output IDs.
-     * @returns An array of corresponding output responses.
+     * Finds output metadata by output ID.
+     * GET /api/core/v3/outputs/{outputId}/metadata
      */
-    async getOutputsIgnoreNotFound(
-        outputIds: OutputId[],
-    ): Promise<OutputResponse[]> {
+    async getOutputMetadata(
+        outputId: OutputId,
+    ): Promise<OutputMetadataResponse> {
         const response = await this.methodHandler.callMethod({
-            name: 'getOutputsIgnoreNotFound',
+            name: 'getOutputMetadata',
             data: {
-                outputIds,
+                outputId,
             },
         });
-        const parsed = JSON.parse(response) as Response<OutputResponse[]>;
-        return plainToInstance(OutputResponse, parsed.payload);
+
+        return JSON.parse(response).payload;
     }
 
     /**
-     * Get the included block of a given transaction.
+     * Finds an output with its metadata by output ID.
+     * GET /api/core/v3/outputs/{outputId}/full
+     */
+    async getOutputWithMetadata(
+        outputId: OutputId,
+    ): Promise<OutputWithMetadataResponse> {
+        const response = await this.methodHandler.callMethod({
+            name: 'getOutputWithMetadata',
+            data: {
+                outputId,
+            },
+        });
+
+        const parsed = JSON.parse(
+            response,
+        ) as Response<OutputWithMetadataResponse>;
```

Comment:
When do we need that stuff exactly?

Reply:
good question ... we might not ... I'll check

Reply:
so, I checked the response and it doesn't seem to make any difference, but someone other than me did this for classes, and probably for a good reason to make sure they are properly instantiated.